etc/bash_completion.d/juju-2.0

```diff
@@ -0,0 +1,357 @@
+# juju-core.bash_completion.sh: dynamic bash completion for juju 2.0 cmdline,
+# from parsed (and cached) juju status output.
+#
+# Author: JuanJo Ciarlante <jjo@canonical.com>
+# Copyright 2016+, Canonical Ltd.
+# License: GPLv3
+#
+# Includes --model and --controller handling:
+#   juju list-models --controller <TAB>
+#   juju switch <TAB>
+#   juju status --model <TAB>
+#   juju ssh --model <TAB> [... will complete with proper model's units/etc ...]
```

Comment:
Did you want to describe other supported completions here as well? (e.g. storage)